core/Pos/Binary/Class/Primitive.hs

```diff
@@ -3,110 +3,135 @@
+                                                   safePut)
+import qualified Data.Serialize                   as Cereal (Get, Put)
+import           Data.Typeable                    (typeRep)
+
+import           Pos.Binary.Class.Core            (Bi(..))
 import           Universum
+import           Serokell.Data.Memory.Units       (Byte)
 
-import qualified Data.ByteString            as BS
-import           Data.Char                  (isAscii)
-import           Data.Reflection            (reifyNat)
-import           Data.SafeCopy              (Contained, SafeCopy (..), contain, safeGet,
-                                             safePut)
-import qualified Data.Serialize             as Cereal (Get, Put)
-import qualified Data.Store.Core            as Store
-import qualified Data.Store.Internal        as Store
-import           Formatting                 (formatToString, int, (%))
-
-import           Pos.Binary.Class.Core      (Bi (..), decodeFull, encode, isolate64Full,
-                                             limitGet)
-import           Pos.Binary.Class.Numbers   (TinyVarInt (..), UnsignedVarInt (..),
-                                             getWord8, putWord8)
-import           Pos.Binary.Class.Store     (Peek, Poke, PokeWithSize (..), Size (..),
-                                             StaticSize (..), putS)
-
-import           Pos.Binary.Class.Instances ()
--- TODO ^ Move Raw in Pos.Util.Util and remove the import.
-
--- | See 'putWord8'.
-putWord8S :: Word8 -> PokeWithSize ()
-putWord8S = putS @Word8
-
--- | See 'putBytes'.
-putBytesS :: ByteString -> PokeWithSize ()
-putBytesS bs = PokeWithSize (BS.length bs) (putBytes bs)
+-- | Serialize a Haskell value to an external binary representation.
+--
+-- The output is represented as a lazy 'BSL.ByteString' and is constructed
+-- incrementally.
+serialize :: Bi a => a -> BSL.ByteString
+serialize = CBOR.Write.toLazyByteString . encode
 
-----------------------------------------------------------------------------
--- Raw
-----------------------------------------------------------------------------
+-- | Serialize a Haskell value to an external binary representation.
+--
+-- The output is represented as a strict 'BS.ByteString'.
+serialize' :: Bi a => a -> BS.ByteString
+serialize' = BSL.toStrict . serialize
 
--- | A wrapper over 'ByteString' for signalling that a bytestring should be
--- processed as a sequence of bytes, not as a separate “entity”. It's used in
--- crypto and binary code.
-newtype Raw = Raw ByteString
-    deriving (Bi, Eq, Ord, Show, Typeable, NFData)
+-- | Deserialize a Haskell value from the external binary representation
+-- (which must have been made using 'serialize' or related function).
+--
+-- /Throws/: @'CBOR.Read.DeserialiseFailure'@ if the given external
+-- representation is invalid or does not correspond to a value of the
+-- expected type.
+deserialize :: Bi a => BSL.ByteString -> a
+deserialize bs0 =
```

Comment:
I don't think that makes any difference. Notice that that is `throw` not `throwIO` and there is no `throwST`. So you're using pure `throw` either way: this way or using `either throw id . deserializeOrFail`